src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -153,14 +147,17 @@ ss::future<> remove_client_if(List& list, Pred pred) {
         }
     }
     for (auto& item : remove) {
-        co_await item.client->stop()
-          .handle_exception([&item](std::exception_ptr ex) {
-              // The stop failed
-              vlog(
-                plog.debug,
-                "Stale client {} stop already happened {}",
-                item.key,
-                ex);
+        co_await item.client_mu
+          ->with([&item]() {
```

Comment:
I see the idea here is to return a mutex with the client, and then you can wait on conditions like stopping all clients before you close the cache. But IMO this is kind of a leaky abstraction, like whats to stop a user from using the pointer and throwing away the mutex?

Maybe a wrapper can prevent this from occurring, something like
```
class wrapped_broker_ptr {
public:
  template<typename Fn>
  auto with_broker(Fn fn) {
    return _mutex.with([fn = std::move(fn)](auto) {
        return fn(_broker);
    }
  }
private:
  ss::lw_shared_ptr<broker> _broker;
};
```
Not sure if this is the right abstraction , but the idea is that no one can now use the resource without it being accessed within the context of the mutex.

Reply:
That's the same point Ben made with this two comments above
https://github.com/redpanda-data/redpanda/pull/6781#discussion_r1017903665
https://github.com/redpanda-data/redpanda/pull/6781#discussion_r1017908622

I understand what both of you are saying but I don't see how that is necessary for merging. This change doesn't affect execution so I see this as a nit.

Reply:
Spoke to rob offline about this. Adding the wrapper is about sticking to our coding standard, so I'll add it in.

Reply:
Done.